pallets/loans/src/entities/pricing/external.rs

```diff
@@ -52,29 +57,44 @@ pub struct ExternalActivePricing<T: Config> {
 
 	/// Outstanding quantity that should be repaid.
 	outstanding_quantity: T::Balance,
+
+	/// Normalized notional debt used to calculate the outstanding debt.
+	normalized_notional: T::Balance,
 }
 
 impl<T: Config> ExternalActivePricing<T> {
 	pub fn new(info: ExternalPricing<T>, pool_id: PoolIdOf<T>) -> Result<Self, DispatchError> {
 		T::PriceRegistry::register_id(&info.price_id, &pool_id)?;
+		T::InterestAccrual::reference_rate(info.interest_rate)?;
 		Ok(Self {
 			info,
 			outstanding_quantity: T::Balance::zero(),
+			normalized_notional: T::Balance::zero(),
 		})
 	}
 
 	pub fn end(self, pool_id: PoolIdOf<T>) -> Result<ExternalPricing<T>, DispatchError> {
 		T::PriceRegistry::unregister_id(&self.info.price_id, &pool_id)?;
+		T::InterestAccrual::unreference_rate(self.info.interest_rate)?;
 		Ok(self.info)
 	}
 
 	pub fn has_debt(&self) -> bool {
-		!self.outstanding_quantity.is_zero()
+		!self.normalized_notional.is_zero()
 	}
 
 	pub fn calculate_debt(&self) -> Result<T::Balance, DispatchError> {
-		let price = self.calculate_price()?;
-		Ok(price.ensure_mul_int(self.outstanding_quantity)?)
+		let now = T::Time::now().as_secs();
+		T::InterestAccrual::calculate_debt(self.info.interest_rate, self.normalized_notional, now)
+	}
+
+	pub fn interest_accrued(&self) -> Result<T::Balance, DispatchError> {
+		let principal = self
```

Comment:
I would only use the term principal for the "to be repaid amount excluding interest", so rather here something like

```suggestion
		let outstanding_notional = self
```